docs/user/unsupported/mbstyle.rst

```diff
@@ -19,14 +24,544 @@ Quick example of module usage (API change):
+  *Optional string*
+
+  If specified, the function will take the specified feature property as an input. See *Zoom Functions and Property Functions* for more information.
+
+  base
+  ''''
+  *Optional number. Default is 1.*
+
+  The exponential base of the interpolation curve. It controls the rate at which the function output increases. Higher values make the output increase more towards the high end of the range. With values close to 1 the output increases linearly.
+
+  type
+  ''''
+  *Optional enum. One of identity, exponential, interval, categorical*
+
+    identity
+    ''''''''
+    functions return their input as their output.
+
+    exponential
+    '''''''''''
+    functions generate an output by interpolating between stops just less than and just greater than the function input. The domain must be numeric.
+
+    interval
+    ''''''''
+    functions return the output value of the stop just less than the function input. The domain must be numeric.
+
+    categorical
+    '''''''''''
+    functions return the output value of the stop equal to the function input.
+
+    default
+    '''''''
+    A value to serve as a fallback function result when a value isn't otherwise available. It is used in the following circumstances:
+
+      * In categorical functions, when the feature value does not match any of the stop domain values.
+      * In property and zoom-and-property functions, when a feature does not contain a value for the specified property.
+      * In identity functions, when the feature value is not valid for the style property (for example, if the function is being used for a circle-color property but the feature property value is not a string or not a valid color).
+      * In interval or exponential property and zoom-and-property functions, when the feature value is not numeric.
+    If no default is provided, the style property's default is used in these circumstances.
+
+    colorSpace
+    ''''''''''
+    *Optional enum. One of rgb, lab, hcl*
+
+    The color space in which colors interpolated. Interpolating colors in perceptual color spaces like LAB and HCL tend to produce color ramps that look more consistent and produce colors that can be differentiated more easily than those interpolated in RGB space.
+
+    *rgb*
+
+    Use the RGB color space to interpolate color values
+
+    *lab*
+
+    Use the LAB color space to interpolate color values.
+
+    *hcl*
+
+    Use the HCL color space to interpolate color values, interpolating the Hue, Chroma, and Luminance channels individually.
+
+    **Zoom Functions** allow the appearance of a map feature to change with map’s zoom level. Zoom functions can be used to create the illusion of depth and control data density. Each stop is an array with two elements: the first is a zoom level and the second is a function output value.
+
+    .. code-block:: json
+
+      {
+        "circle-radius": {
+          "stops": [
+            // zoom is 5 -> circle radius will be 1px
```

Comment:
I am not sure if we support comments, if we do not support comments then we should not use it in examples.